core/templates/base-components/base-content.component.ts

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the Base Transclusion Component.
+ */
+
+import { Component, Input } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import constants from 'assets/constants';
+import { BottomNavbarStatusService } from 'services/bottom-navbar-status.service';
+import { UrlService } from 'services/contextual/url.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { KeyboardShortcutService } from 'services/keyboard-shortcut.service';
+import { LoaderService } from 'services/loader.service';
+import { PageTitleService } from 'services/page-title.service';
+import { SidebarStatusService } from 'services/sidebar-status.service';
+import { BackgroundMaskService } from 'services/stateful/background-mask.service';
+
+@Component({
+  selector: 'oppia-base-content',
+  templateUrl: './base-content.component.html'
+})
+export class BaseContentComponent {
+  @Input() backButtonShown: boolean;
+  loadingMessage: string = '';
+  mobileNavOptionsAreShown: boolean = false;
+  iframed: boolean;
+  DEV_MODE = constants.DEV_MODE;
+  getHeaderText: () => string;
+  getSubheaderText: () => string;
+
+  constructor(
+    private windowRef: WindowRef,
+    private backgroundMaskService: BackgroundMaskService,
+    private bottomNavbarStatusService: BottomNavbarStatusService,
+    private keyboardShortcutService: KeyboardShortcutService,
+    private loaderService: LoaderService,
+    private pageTitleService: PageTitleService,
+    private sidebarStatusService: SidebarStatusService,
+    private urlService: UrlService,
+  ) {}
+
+  ngOnInit(): void {
+    if (this.windowRef.nativeWindow.location.hostname ===
+      'oppiaserver.appspot.com') {
```

Comment:
Can we separate this into a function for readability? Smth like `isOppiaAppspotServer()`

Reply:
done